### ISSUE TITLE
Windows nightly crashes on startup for some users

### ISSUE THREAD

**Issue by jdm** [servo/servo#22315]
@samal84 reported in https://github.com/servo/servo/issues/21820#issuecomment-442920783 that the Windows nightly installer produces an installation that crashes at startup. @samal84 could you attach a screenshot of the error that you see?

**Comment by atouchet** [comment]
I suspect there are still issues with the GStreamer installation on some systems. I will try to look into this.

**Comment by atouchet** [comment]
@ferjm do you have any ideas about that issue with the extra backslash?

**Comment by atouchet** [comment]
I tried testing this out by uninstalling Servo and GStreamer from my system and doing a fresh install from scratch. I can confirm that I run into issues when trying to launch Servo using just the installer from servo.org. When I launch Servo I get this error:

![Error](https://i.imgur.com/Az2hnPN.png)

GStreamer installs to my G: drive and these are the environment variables:

![Environment variables](https://i.imgur.com/kfdwjqT.png)
*Other info edited out

Previously if I manually installed GStreamer to C:/gstreamer that would be enough to get Servo to run. I suspect this issue may only occur on setups with multiple drives available.

**Comment by kavika13** [comment]
I have one drive, and I get the same error on launch.
`GSTREAMER_1_0_ROOT_X86` points to `C:\gstreamer\1.0\x86\`
The files seem to have unpacked to `C:\gstreamer\1.0\x86\` as well.

I haven't installed any previous versions. I'm running Win10 Pro x64.

**Comment by atouchet** [comment]
I don't think #22322 solves the error. I just tested the latest nightly and get the exact same issue still.

**Comment by atouchet** [comment]
@ferjm I am fairly certain the error is due to the GStreamer files. If I manually install `gstreamer-1.0-x86_64-1.14.4.msi` and `gstreamer-1.0-devel-x86_64-1.14.4.msi` to `G:\gstreamer` I get the same error screen as before and Servo will not start. However if I manually install those GStreamer files to `C:\gstreamer` Servo then runs fine.

**Comment by atouchet** [comment]
#21968 might help with this.

**Comment by qqgg231** [comment]
<img width="300" alt="screenshot_6" src="https://user-images.githubusercontent.com/11590960/49686645-bb3cf300-faf7-11e8-8168-9cb75efe452a.png">

i get this on startup

it says this dll not found

**Comment by Mardeg** [comment]
Getting the same error on today's build. I ran dependencywalker on servo.exe and got:
--------------------------------------------------------------------------------
Operating System: Microsoft Windows NT/2000/XP/2003/Vista based Professional (64-bit), version 6.02.9200
Program Executable: c:\program files\mozilla research\servo tech demo\SERVO.EXE
Program Arguments: 
Starting Directory: C:\Program Files\Mozilla Research\Servo Tech Demo\
Search Path: C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\gstreamer\1.0\x86\bin;1.0\x86_64\bin;C:\Users\User1\AppData\Local\Microsoft\WindowsApps;

Options Selected:
     Simulate ShellExecute by inserting any App Paths directories into the PATH environment variable.
     Log DllMain calls for process attach and process detach messages.
     Log DllMain calls for all other messages, including thread attach and thread detach.
     Hook the process to gather more detailed dependency information.
     Log LoadLibrary function calls.
     Log GetProcAddress function calls.
     Log first chance exceptions.
     Log debug output messages.
     Use full paths when logging file names.
     Automatically open and profile child processes.
--------------------------------------------------------------------------------

Started "c:\program files\mozilla research\servo tech demo\SERVO.EXE" (process 0x5170) at address 0x00007FF64E510000.  Successfully hooked module.
Loaded "c:\windows\system32\NTDLL.DLL" at address 0x00007FFA8C830000.  Successfully hooked module.
Loaded "c:\windows\system32\KERNEL32.DLL" at address 0x00007FFA8AA30000.  Successfully hooked module.
Loaded "c:\windows\system32\KERNELBASE.DLL" at address 0x00007FFA88FF0000.  Successfully hooked module.
DllMain(0x00007FFA88FF0000, DLL_PROCESS_ATTACH, 0x0000000000000000) in "c:\windows\system32\KERNELBASE.DLL" called.
DllMain(0x00007FFA88FF0000, DLL_PROCESS_ATTACH, 0x0000000000000000) in "c:\windows\system32\KERNELBASE.DLL" returned 1 (0x1).
DllMain(0x00007FFA8AA30000, DLL_PROCESS_ATTACH, 0x0000000000000000) in "c:\windows\system32\KERNEL32.DLL" called.
DllMain(0x00007FFA8AA30000, DLL_PROCESS_ATTACH, 0x0000000000000000) in "c:\windows\system32\KERNEL32.DLL" returned 1 (0x1).
Injected "c:\users\user1\downloads\dependencywalker\DEPENDS.DLL" at address 0x00000000623E0000.
DllMain(0x00000000623E0000, DLL_PROCESS_ATTACH, 0x0000000000000000) in "c:\users\user1\downloads\dependencywalker\DEPENDS.DLL" called.
DllMain(0x00000000623E0000, DLL_PROCESS_ATTACH, 0x0000000000000000) in "c:\users\user1\downloads\dependencywalker\DEPENDS.DLL" returned 1 (0x1).
Loaded "c:\windows\system32\USER32.DLL" at address 0x00007FFA8AE00000.  Successfully hooked module.
Loaded "c:\windows\system32\WIN32U.DLL" at address 0x00007FFA88EF0000.  Successfully hooked module.
Loaded "c:\program files\mozilla research\servo tech demo\LIBCRYPTOMD.DLL" at address 0x00007FFA55EB0000.  Successfully hooked module.
Loaded "c:\program files\mozilla research\servo tech demo\LIBSSLMD.DLL" at address 0x00007FFA70D50000.  Successfully hooked module.
Loaded "c:\windows\system32\GDI32.DLL" at address 0x00007FFA8AC70000.  Successfully hooked module.
Loaded "c:\windows\system32\WS2_32.DLL" at address 0x00007FFA8A7C0000.  Successfully hooked module.
Loaded "c:\windows\system32\UCRTBASE.DLL" at address 0x00007FFA89980000.  Successfully hooked module.
Loaded "c:\windows\system32\RPCRT4.DLL" at address 0x00007FFA8C5D0000.  Successfully hooked module.
Loaded "c:\windows\system32\VCRUNTIME140.DLL" at address 0x00007FFA78CD0000.  Successfully hooked module.
Loaded "c:\windows\system32\ADVAPI32.DLL" at address 0x00007FFA89D70000.  Successfully hooked module.
Loaded "c:\windows\system32\MSVCRT.DLL" at address 0x00007FFA89E40000.  Successfully hooked module.
Loaded "c:\windows\system32\GDI32FULL.DLL" at address 0x00007FFA88C00000.  Successfully hooked module.
Loaded "c:\windows\system32\SECHOST.DLL" at address 0x00007FFA8B130000.  Successfully hooked module.
Loaded "c:\windows\system32\MSVCP_WIN.DLL" at address 0x00007FFA88DA0000.  Successfully hooked module.
Loaded "c:\windows\system32\CRYPT32.DLL" at address 0x00007FFA89AD0000.  Successfully hooked module.
Loaded "c:\windows\system32\MSASN1.DLL" at address 0x00007FFA88B60000.  Successfully hooked module.
Loaded "c:\windows\system32\COMDLG32.DLL" at address 0x00007FFA8B040000.  Successfully hooked module.
Loaded "c:\windows\system32\COMBASE.DLL" at address 0x00007FFA8A030000.  Successfully hooked module.
Loaded "c:\windows\system32\BCRYPTPRIMITIVES.DLL" at address 0x00007FFA88F70000.  Successfully hooked module.
Loaded "c:\windows\system32\SHCORE.DLL" at address 0x00007FFA8AF90000.  Successfully hooked module.
Loaded "c:\windows\system32\SHLWAPI.DLL" at address 0x00007FFA89EF0000.  Successfully hooked module.
Loaded "c:\windows\system32\SHELL32.DLL" at address 0x00007FFA8B190000.  Successfully hooked module.
Loaded "c:\windows\winsxs\amd64_microsoft.windows.common-controls_6595b64144ccf1df_5.82.17134.472_none_f9558ad13092c43c\COMCTL32.DLL" at address 0x00007FFA7A1F0000.  Successfully hooked module.
Loaded "c:\windows\system32\CFGMGR32.DLL" at address 0x00007FFA89A80000.  Successfully hooked module.
Loaded "c:\windows\system32\WINDOWS.STORAGE.DLL" at address 0x00007FFA89270000.  Successfully hooked module.
Loaded "c:\windows\system32\KERNEL.APPCORE.DLL" at address 0x00007FFA88B40000.  Successfully hooked module.
Loaded "c:\windows\system32\PROFAPI.DLL" at address 0x00007FFA88B80000.  Successfully hooked module.
Loaded "c:\windows\system32\POWRPROF.DLL" at address 0x00007FFA88BA0000.  Successfully hooked module.
Loaded "c:\windows\system32\FLTLIB.DLL" at address 0x00007FFA88BF0000.  Successfully hooked module.
Loaded "c:\windows\system32\OLE32.DLL" at address 0x00007FFA8ACA0000.  Successfully hooked module.
Loaded "c:\gstreamer\1.0\x86\bin\LIBGSTVIDEO-1.0-0.DLL" at address 0x000000006D400000.  Successfully hooked module.
Unloaded "c:\gstreamer\1.0\x86\bin\LIBGSTVIDEO-1.0-0.DLL" at address 0x000000006D400000.
Loaded "c:\gstreamer\1.0\x86\bin\LIBGSTPLAYER-1.0-0.DLL" at address 0x0000000066D00000.  Successfully hooked module.
Loaded "c:\gstreamer\1.0\x86\bin\LIBGSTAPP-1.0-0.DLL" at address 0x00000000703C0000.  Successfully hooked module.
Unloaded "c:\gstreamer\1.0\x86\bin\LIBGSTPLAYER-1.0-0.DLL" at address 0x0000000066D00000.
Unloaded "c:\gstreamer\1.0\x86\bin\LIBGSTAPP-1.0-0.DLL" at address 0x00000000703C0000.
Loaded "c:\gstreamer\1.0\x86\bin\LIBGSTAUDIO-1.0-0.DLL" at address 0x0000000064B80000.  Successfully hooked module.
Unloaded "c:\gstreamer\1.0\x86\bin\LIBGSTAUDIO-1.0-0.DLL" at address 0x0000000064B80000.
Exited "c:\program files\mozilla research\servo tech demo\SERVO.EXE" (process 0x5170) with code -1073741701 (0xC000007B).

Error: At least one required implicit or forwarded dependency was not found.
Error: At least one module has an unresolved import due to a missing export function in an implicitly dependent module.
Error: Modules with different CPU types were found.
Warning: At least one delay-load dependency module was not found.
Warning: At least one module has an unresolved import due to a missing export function in a delay-load dependent module.

[servoerrors.txt](https://github.com/servo/servo/files/2718232/servoerrors.txt)

**Comment by jdm** [comment]
If I'm reading that log correctly, it looks like the soudermonkey symbols are a problem.

**Comment by Mart-Bogdan** [comment]
For me it's crashing on start without displaying any error message, and exit code is ` -1073741701 (0xc000007b)`

I believe it's panicking, but if someone could provide console build, not windows build, some errors could show up in console.

**Comment by sheddup** [comment]
is the reason for crashing understood yet?

**Comment by atouchet** [comment]
I am no longer able to reproduce the crashes with either the ZIP or installer version of Servo after #21968 and #22811 were merged. Could someone else confirm that this problem is fixed for them or not?

**Comment by Mardeg** [comment]
It now brings up a blank window and immediately vanishes, presumably silently crashing with no dialog. Dependencywalker shows one less error than before so I guess that's progress.

Error: At least one required implicit or forwarded dependency was not found.
Error: At least one module has an unresolved import due to a missing export function in an implicitly dependent module.
Warning: At least one delay-load dependency module was not found.
Warning: At least one module has an unresolved import due to a missing export function in a delay-load dependent module.

The "Modules with different CPU types were found." error is no longer listed.

No idea if it's useful but inside servo.exe it lists

c:\program files\mozilla research\servo tech demo\LIBSSLMD.DLL

and inside that is the following with functions that have "not bound" listed for their entry point:

c:\program files\mozilla research\servo tech demo\VCRUNTIME140.DLL

API-MS-WIN-CRT-TIME-L1-1-0.DLL
_time64

API-MS-WIN-CRT-STRING-L1-1-0.DLL
_stricmp
_strnicmp
strncmp

API-MS-WIN-CRT-UTILITY-L1-1-0.DLL
qsort

**Comment by atouchet** [comment]
@Mardeg what version of Windows are you using?

**Comment by kavika13** [comment]
I installed the nightly yesterday, directly over the other one, without uninstalling. It is now launching correctly for me, and not crashing when I try a few sites.

**Comment by atouchet** [comment]
I think this can be closed now as it sounds like the issue has been fixed.

@Mardeg if you are still having issues with running the Windows build could you please open a new issue report as that doesn't appear to be the same issue that others were running into.
